bpf/bpf_overlay.c

```diff
@@ -755,19 +756,22 @@ int cil_to_overlay(struct __ctx_buff *ctx)
 	}
 #endif
 
-#ifdef ENABLE_NODEPORT
-	if (ctx_snat_done(ctx)) {
-		ret = CTX_ACT_OK;
-		goto out;
-	}
-
 	/* This must be after above ctx_snat_done, since the MARK_MAGIC_CLUSTER_ID
 	 * is a super set of the MARK_MAGIC_SNAT_DONE. They will never be used together,
 	 * but SNAT check should always take presedence.
 	 */
 #ifdef ENABLE_CLUSTER_AWARE_ADDRESSING
 	cluster_id = ctx_get_cluster_id_mark(ctx);
 #endif
+
+	ctx_set_overlay_mark(ctx);
```

Comment:
Just as a sanity check, this occurs **after** vxlan encap right?

Reply:
This **is** the encap :smile:.

`to-overlay ` processes the inner packet, then `vxlan.ko` adds the outer headers. The mark survives.

Reply:
Traced this out with @julianwiedmann 

This is only a scenario if the packet is going cross net-namespace. Resolving.